### ISSUE TITLE
Email filter doesnt work

### ISSUE THREAD

**Issue by 4n0n1mo** [webmin/webmin#1668]
I'm filtering emails with a rule that throws away emails that the subject contains "ncia para sua conta banc".

It's not throwing them out, it's going to the inbox folder, but when I click the button "Show Matching Email In Folder: " in Inbox folder, its showing me the emails that it were supposed to be filtered out.

**Comment by jcameron** [comment]
Is this filter being setup under a mailbox in Usermin?

**Comment by 4n0n1mo** [comment]
> Is this filter being setup under a mailbox in Usermin?

Yes, I'm setting this up in Mail - Usermin, "Email Filters" menu option

**Comment by jcameron** [comment]
Are there any accented or non-ASCII characters in the filter, or in the email subject lines?

**Comment by 4n0n1mo** [comment]
There is accented characters in the email subject, but not in the characters that I've used in the filter "ncia para sua conta banc", it works when I click to test the filter, but the email messages still not being throwed away.

**Comment by jcameron** [comment]
Ah, I wonder if this is due to the encoding of the subject line. If you view a raw message, what does the `Subject:` header contain?

**Comment by philmck** [comment]
I'm experiencing this as well. Here's a screenshot showing a "throwaway" action preceding a "save in folder" filter. The "save in folder" happens but the throwaway doesn't. If I go to Usermin and run the filter on the "webmaster" folder it correctly matches the messages I wanted to throw away.

![filter screenshot](https://user-images.githubusercontent.com/293914/214612913-58ab8995-c7f3-4135-8a9e-cef602e77d9a.jpg)

I'm pretty sure this is indeed a character encoding issue. Here's one of the affected subject headers:

> Subject: =?us-ascii?Q?[Warleigh_Hall_Press]_Some_plugins_were_automat?=
>  =?us-ascii?Q?ically_updated?=

I have no idea why WordPress is messing up the subject encoding so badly. Nor do I have any idea why the filtering works after the message is delivered to the folder but not before. But it's clearly a fixable bug.

**Comment by jcameron** [comment]
Unfortunately this is hard for Webmin to fix, as it is `procmail` that does the email filtering and it doesn't decode subject headers like that. Long-term we'll need to switch to another email filter, like seive..

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.

**Comment by iliaross** [comment]
To be clear, this is caused by the current Usermin email filter backend relying on `procmail` for delivery.

Procmail is a very old mail filtering tool. It dates back to 1990, so it is over 35 years old now. It has been supported and used for decades, and even its current upstream describes version 3.24 as a 2022 release after “two decades of fixes.” Yet, its design predates modern MIME header handling, and it evaluates the raw message headers during delivery.

Usermin can decode MIME-encoded subject headers when displaying or searching messages, which is why “Show Matching Email” may find them. Procmail does not decode MIME-encoded or folded Subject headers before matching, so filters against decoded subject text can fail during actual delivery even though they match later in the UI.

Fixing this properly would require moving these filters to a backend that understands decoded mail headers, such as Sieve, rather than patching around `procmail`.

Closing this as not planned.
